fleet_vehicle_stock/tests/test_fleet_vehicle.py

```diff
@@ -0,0 +1,113 @@
+
+
+class TestFleetVehicle(TransactionCase):
+    def setUp(self):
+        super(TestFleetVehicle, self).setUp()
+        self.vehicle = self.env["fleet.vehicle"]
+        self.vehicle_model = self.env["fleet.vehicle.model"]
+        self.brand = self.env.ref("fleet.brand_opel")
+        self.stock_location = self.env.ref("stock.stock_location_customers")
+        self.product = self.env["product.template"]
+
+        self.vehicle_model = self.vehicle_model.create(
+            {"name": "Test Vehicle Model", "brand_id": self.brand.id}
+        )
+
+        product1 = self.env["product.product"].create(
+            {
+                "name": "Product A",
+                "type": "product",
+                "fleet_vehicle_model_id": self.vehicle_model.id,
+                "tracking": "serial",
+            }
+        )
+
+        lot1 = self.env["stock.production.lot"].create(
+            {
+                "name": "serial1",
+                "product_id": product1.id,
+                "company_id": self.env.user.company_id.id,
+            }
+        )
+
+        lot2 = self.env["stock.production.lot"].create(
+            {
+                "name": "serial2",
+                "product_id": product1.id,
+                "company_id": self.env.user.company_id.id,
+            }
+        )
+
+        self.env["stock.quant"].create(
+            {
+                "product_id": product1.id,
+                "location_id": self.stock_location.id,
+                "quantity": 1.0,
+                "lot_id": lot1.id,
+            }
+        )
+
+        self.vehicle = self.vehicle.create(
+            {
+                "model_id": self.vehicle_model.id,
+                "product_id": product1.id,
+                "lot_id": lot1.id,
+                "current_stock_location_id": self.stock_location.id,
+            }
+        )
+
+        self.vehicle_without_quant = self.vehicle.create(
+            {
+                "model_id": self.vehicle_model.id,
+                "product_id": product1.id,
+                "lot_id": lot2.id,
+            }
+        )
+
+        self.vehicle_without_lot_id = self.vehicle.create(
+            {"model_id": self.vehicle_model.id, "product_id": product1.id}
+        )
+
+        self.product2 = self.product.create(
+            {
+                "name": "Product B",
+                "type": "product",
+                "fleet_vehicle_model_id": self.vehicle_model.id,
+                "tracking": "serial",
+            }
+        )
+
+    def test_onchange_product(self):
+        vehicle = self.vehicle
+        vehicle._onchange_product()
+        self.assertFalse(vehicle.current_stock_location_id)
+
+    def test_onchange_model(self):
+        vehicle = self.vehicle
+        vehicle._onchange_model()
+        self.assertTrue(vehicle.product_id == vehicle.model_id.product_id)
+
+    def test_compute_current_stock_loc_id(self):
+        vehicle = self.vehicle
+        vehicle._compute_current_stock_loc_id()
```

Comment:
```suggestion
```
You shouldn't have to call the compute method directly.
You should rather test that it's recomputed when you want to (e.g.: changing one of the fields in its `api.depends`)